src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1107,29 +1107,40 @@ impl Cage {
             None => STARTINGFD,
         };
 
+        if start_fd == fd { return start_fd; } //if the file descriptors are equal (which is not plausible here), return
```

Comment:
Do we need this check as there is already a check for `new_fd == old_fd` in the `dup2_syscall`?

Reply:
As I commented, this would not happen, I just leave it here for redundancy, but yeah, maybe I'll delete this

Reply:
dup and dup2 are separate, i think we need to ad this back